### PR TITLE
Removing minimum temperature for HPWH operation

### LINKED ISSUE #9758: Reduce minimum ambient temperature for HPWH operation
Issue overview
--------------
When modeling HPWHs, E+ only allows the minimum temperature for HP operation to go as low as -5 C. This limit seems to be hard coded into the water heater code (see https://github.com/NREL/EnergyPlus/blob/0b1a165e0aa6b1f9fc88991c987673fdcd366dad/src/EnergyPlus/WaterThermalTanks.cc#L1537-L1544), so there doesn't seem to be any way to work around this. There are several products on the market that can operate at much lower temperatures than -5 C, such as the SanCO2 which can operate below -30 C (https://www.eco2waterheater.com/_files/ugd/e88920_293f399a2a3a432396b1805a76f21c80.pdf). To be able to properly model these products, we'd need to lower this limit. My suggestion would be to set this to at least -40 C for products like the SanCO2. It'd also be nice to move this lower limit to the .idd rather than hard coding in the event that someone needs to lower this further for some new product with an even broader operating range.

### Checklist
Add to this list or remove from it as applicable.  This is a simple templated set of guidelines.
 - [ ] Defect file added (list location of defect file here)
 - [ ] Ticket added to Pivotal for defect (development team task)
 - [ ] Pull request created (the pull request will have additional tasks related to reviewing changes that fix this defect)

### PR DESCRIPTION
**Pull request by nmerket**
Pull request overview
---------------------
 - Fixes #9758

As described in the issue, this restriction is arbitrary and not relevant to newer heat pump water heaters. 

**NOTE: ENHANCEMENTS MUST FOLLOW A SUBMISSION PROCESS INCLUDING A FEATURE PROPOSAL AND DESIGN DOCUMENT PRIOR TO SUBMITTING CODE**

### Pull Request Author
Add to this list or remove from it as applicable.  This is a simple templated set of guidelines.
 - [x] Title of PR should be user-synopsis style (clearly understandable in a standalone changelog context)
 - [x] Label the PR with at least one of: Defect, Refactoring, NewFeature, Performance, and/or DoNoPublish
 - [x] ~~Pull requests that impact EnergyPlus code must also include unit tests to cover enhancement or defect repair~~
 - [x] Author should provide a "walkthrough" of relevant code changes using a GitHub code review comment process
 - [x] ~~If any diffs are expected, author must demonstrate they are justified using plots and descriptions~~
 - [x] ~~If changes fix a defect, the fix should be demonstrated in plots and descriptions~~
 - [x] ~~If any defect files are updated to a more recent version, upload new versions here or on DevSupport~~
 - [x] ~~If IDD requires transition, transition source, rules, ExpandObjects, and IDFs must be updated, and add IDDChange label~~
 - [x] ~~If structural output changes, add to output rules file and add OutputChange label~~
 - [x] ~~If adding/removing any LaTeX docs or figures, update that document's CMakeLists file dependencies~~

### Reviewer
This will not be exhaustively relevant to every PR.
 - [ ] Perform a Code Review on GitHub
 - [ ] If branch is behind develop, merge develop and build locally to check for side effects of the merge
 - [ ] If defect, verify by running develop branch and reproducing defect, then running PR and reproducing fix
 - [ ] If feature, test running new feature, try creative ways to break it
 - [ ] CI status: all green or justified
 - [ ] Check that performance is not impacted (CI Linux results include performance check)
 - [ ] Run Unit Test(s) locally
 - [ ] Check any new function arguments for performance impacts
 - [ ] Verify IDF naming conventions and styles, memos and notes and defaults
 - [ ] If new idf included, locally check the err file and other outputs

### CODE REVIEW

**Comment by mjwitte** [comment]
@nmerket The idd has no limits for this input field, so that's ok. But what about the limits for the max inlet temp field? Are these appropriate or should they be removed from the IDD?
```
WaterHeater:HeatPump:WrappedCondenser, (similar for WaterHeater:HeatPump:PumpedCondenser)
  N5 , \field Minimum Inlet Air Temperature for Compressor Operation
       \type real
       \units C
       \default 10
       \note Heat pump compressor will not operate when the inlet air dry-bulb temperature
       \note is below this value.
  N6 , \field Maximum Inlet Air Temperature for Compressor Operation
       \type real
       \units C
       \default 48.88888888889
       \minimum 26
       \maximum 94
       \note Heat pump compressor will not operate when the inlet air dry-bulb temperature
       \note is above this value.
```

**Comment by nmerket** [comment]
@jmaguire1 @shorowit: see @mjwitte's question above.

**Comment by nmerket** [comment]
@mjwitte I think those maximum limits are reasonable and should remain. There's just been some movement in the market to have heat pumps that can operate in colder temperatures, which is necessitates this change.

**Comment by jmaguire1** [comment]
I've never seen products that can run above 120 F. Maybe they exist, but it's way more common to run into issues with low temperature rather than high temperatures, so keeping the upper temperature limit as it is seems reasonable to me.

File: src/EnergyPlus/WaterThermalTanks.cc [hunk 1]
```diff
@@ -1536,12 +1536,6 @@ bool getHPWaterHeaterInput(EnergyPlusData &state)
 
         // Minimum Inlet Air Temperature for Compressor Operation
         HPWH.MinAirTempForHPOperation = hpwhNumeric[4 + nNumericOffset];
-        if (HPWH.MinAirTempForHPOperation < -5) {
-            ShowWarningError(state,
-                             state.dataIPShortCut->cCurrentModuleObject + "=\"" + HPWH.Name +
-                                 "\": minimum inlet air temperature for heat pump compressor operation must be greater than or equal to -5 C.");
-            ShowContinueError(state, format("...Minimum inlet air temperature = {:.1T}", hpwhNumeric[4 + nNumericOffset]));
-        }
```
Comment (nmerket) [inline src/EnergyPlus/WaterThermalTanks.cc:1544]: This is it. Removing this check/validation. It wasn't in the idd anyway. There are heat pumps that operate at a lower temperature than this now.

Comment (Myoldmopar) [reply src/EnergyPlus/WaterThermalTanks.cc:1544]: Super reasonable.  Thanks @nmerket

**Review by Myoldmopar — APPROVED** [review]

**Comment by Myoldmopar** [comment]
Simple change, 100% pass rate on CI, merging.  Thanks @nmerket

**Pull request merged**